tensorflow_addons/image/cutout_ops.py

```diff
@@ -139,26 +152,33 @@ def cutout(
             ],
             [1, 0],
         )
-        masks = tf.TensorArray(images.dtype, 0, dynamic_size=True)
-        for i in tf.range(tf.shape(cutout_shape)[0]):
+
+        def fn(i):
             padding_dims = [
                 [lower_pads[i], upper_pads[i]],
                 [left_pads[i], right_pads[i]],
             ]
             mask = tf.pad(
-                tf.zeros(cutout_shape[i], dtype=images.dtype),
+                tf.zeros(cutout_shape[i], dtype=tf.bool),
                 padding_dims,
-                constant_values=1,
+                constant_values=True,
             )
-            masks = masks.write(i, mask)
-
-        mask_4d = tf.expand_dims(masks.stack(), -1)
-        mask = tf.tile(mask_4d, [1, 1, 1, tf.shape(images)[-1]])
+            return mask
+
+        mask = tf.map_fn(
```

Comment:
Is vectorized_map a better choice?

Reply:
it somehow traces the graph multiple times, see the colab link above.

Reply:
Yeah, map_fn is more suitable